examples/paho-mqtt/main.c

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (C) 2019 Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file        main.c
+ * @brief       Example using MQTT Paho package from RIOT
+ *
+ * @author      Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include "xtimer.h"
+#include "shell.h"
+#include "thread.h"
+#include "mutex.h"
+#include "mqtt.h"
+#include "MQTTClient.h"
+
+#define BUF_SIZE                        1024
+#define MQTT_VERSION_v311               4       /* MQTT v3.1.1 version is 4 */
+#define COMMAND_TIMEOUT_MS              4000
+
+#define DEFAULT_MQTT_PORT               1883    /**< Default MQTT port */
+#define DEFAULT_KEEPALIVE_SEC           10      /**< Keepalive timeout in seconds */
+#define DEFAULT_MQTT_USER               "riot"
```

Comment:
Does it work if ones uses an string for the user in an anonymous connection ?

Reply:
yes it does... mine is set as anonymous and even though passing userid